framework/include/userobjects/JSONFileReader.h

```diff
@@ -36,30 +36,94 @@ class JSONFileReader : public GeneralUserObject
    * @param scalar_name the name of the desired scalar
    * @param scalar reference to the scalar that will be set with the scalar in the JSON
    */
-  void getScalar(const std::string & scalar_name, Real & scalar) const;
+  template <typename T>
+  void getScalar(const std::string & scalar_name, T & scalar) const
```

Comment:
can we "getScalar" for a string or does that sound weird.

if so, should it be getSingleton ?

Reply:
I think that sounds even weirder 🤔

What about getValue and getVectorValue?

Reply:
haha
yes that sounds good